### PR TITLE
Make signTransaction method immutable

### LINKED ISSUE #228: Make signTransaction method immutable
## Context / issue

`signTransaction()` method modifies the param that is received.

## Proposed solution

It's a better practice that signing a transaction returns a new transaction object without mutating the passed argument.

### PR DESCRIPTION
**Pull request by germartinez**
## What it solves
Resolves #228

### CODE REVIEW

File: packages/safe-core-sdk/README.md [hunk 1]
```diff
@@ -108,7 +108,7 @@ Before executing this transaction, it must be signed by the owners and this can
 The `owner1` account signs the transaction off-chain.
 
 ```js
-const owner1Signature = await safeSdk.signTransaction(safeTransaction)
+const signedSafeTransaction = await safeSdk.signTransaction(safeTransaction)
 ```
 
 Because the signature is off-chain, there is no interaction with the contract and the signature becomes available at `safeTransaction.signatures`.
```
Comment (usame-algan) [inline packages/safe-core-sdk/README.md:114]: Should we rephrase this as well to include `signedSafeTransaction`?

Comment (germartinez) [reply packages/safe-core-sdk/README.md:114]: True. Good catch, thanks!

**Review by usame-algan — APPROVED** [review]
